### PR TITLE
Fix some broken links

### PR DESCRIPTION
**Pull request by cwhanse**
- ~~[ ] Closes #xxxx~~
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - ~~[ ] Tests added~~
 - ~~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~~
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~~[ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~~
 - [X] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

More link repairs in response to #1794

### CODE REVIEW

File: pvlib/temperature.py [hunk 1]
```diff
@@ -362,8 +362,9 @@ def pvsyst_cell(poa_global, temp_air, wind_speed=1.0, u_c=29.0, u_v=0.0,
 
     References
     ----------
-    .. [1] "PVsyst 6 Help", Files.pvsyst.com, 2018. [Online]. Available:
-       http://files.pvsyst.com/help/index.html. [Accessed: 10- Dec- 2018].
+    .. [1] "PVsyst 7 Help", [Online]. Available:
+       https://www.pvsyst.com/help/index.html?thermal_loss.htm.
```
Comment (AdamRJensen) [inline pvlib/temperature.py:366]:
```suggestion
       https://www.pvsyst.com/help/index.html?thermal_loss.htm.
```

**Review by AdamRJensen — APPROVED** [review]

File: pvlib/iotools/srml.py [hunk 2]
```diff
@@ -287,10 +287,10 @@ def get_srml(station, start, end, filetype='PO', map_variables=True,
     References
     ----------
     .. [1] University of Oregon Solar Radiation Measurement Laboratory
-       `http://solardat.uoregon.edu/ <http://solardat.uoregon.edu/>`_
+       `http://solardata.uoregon.edu/`_
     .. [2] Station ID codes - Solar Radiation Measurement Laboratory
-       `http://solardat.uoregon.edu/StationIDCodes.html
-       <http://solardat.uoregon.edu/StationIDCodes.html>`_
+       `http://solardata.uoregon.edu/StationIDCodes.html
+       <http://solardata.uoregon.edu/StationIDCodes.html>`_
```
[the remark this answers is not in the document]
Comment (kandersolar) [reply pvlib/iotools/srml.py]: I don't think this formatting is what we want.  It renders like this:

image

If we just want the raw URL to be displayed as the link text, then no markup is needed.  See the `iam.ashrae` docstring for example.

File: pvlib/iotools/srml.py [hunk 3]
```diff
@@ -287,10 +287,9 @@ def get_srml(station, start, end, filetype='PO', map_variables=True,
     References
     ----------
     .. [1] University of Oregon Solar Radiation Measurement Laboratory
-       `http://solardat.uoregon.edu/ <http://solardat.uoregon.edu/>`_
+       http://solardata.uoregon.edu/_
```
Comment (kandersolar) [inline pvlib/iotools/srml.py:290]:
```suggestion
       http://solardata.uoregon.edu/
```

File: pvlib/iotools/srml.py [hunk 4]
```diff
@@ -287,10 +287,9 @@ def get_srml(station, start, end, filetype='PO', map_variables=True,
...
     .. [2] Station ID codes - Solar Radiation Measurement Laboratory
-       `http://solardat.uoregon.edu/StationIDCodes.html
-       <http://solardat.uoregon.edu/StationIDCodes.html>`_
+       http://solardata.uoregon.edu/StationIDCodes.html_
```
Comment (kandersolar) [inline pvlib/iotools/srml.py:292]:
```suggestion
       http://solardata.uoregon.edu/StationIDCodes.html
```

**Review by kandersolar — APPROVED** [review]

**Comment by kandersolar** [comment]
Thanks @cwhanse and @echedey-ls!

**Pull request merged**